pkg/trustedresources/verifier/verifier.go

```diff
@@ -0,0 +1,192 @@
+// verifier using the provided hash algorithm
+func fromKeyRef(ctx context.Context, keyRef string, hashAlgorithm crypto.Hash, k8s kubernetes.Interface) (verifier signature.Verifier, err error) {
+	var raw []byte
+	if strings.HasPrefix(keyRef, keyReference) {
+		v, err := fromSecret(ctx, keyRef, hashAlgorithm, k8s)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get verifier from secret: %w", err)
+		}
+		return v, nil
+	}
+	raw, err = os.ReadFile(filepath.Clean(keyRef))
+	if err != nil {
+		return nil, fmt.Errorf("%w: %v", ErrorFailedLoadKeyFile, err)
+	}
+	v, err := fromData(raw, hashAlgorithm)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get verifier from data: %w", err)
+	}
+	return v, nil
+}
+
+// fromSecret fetches the public key from SecretRef and returns the verifier
+// hashAlgorithm is provided to determine the hash algorithm of the key
+func fromSecret(ctx context.Context, secretRef string, hashAlgorithm crypto.Hash, k8s kubernetes.Interface) (signature.Verifier, error) {
+	if strings.HasPrefix(secretRef, keyReference) {
+		s, err := keyPairSecret(ctx, secretRef, k8s)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get secret: %w", err)
+		}
+		// only 1 public key should be in the secret
+		if len(s.Data) == 0 {
+			return nil, fmt.Errorf("secret %q contains no data %w", secretRef, errorEmptySecretData)
+		}
+		if len(s.Data) > 1 {
+			return nil, fmt.Errorf("secret %q contains multiple data entries, only one is supported. %w", secretRef, errorMultipleSecretData)
+		}
+		for _, raw := range s.Data {
+			v, err := fromData(raw, hashAlgorithm)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get verifier from secret data: %w", err)
+			}
+			return v, nil
+		}
+	}
+	return nil, fmt.Errorf("%w: secretRef %v is invalid", errorK8sSpecificationInvalid, secretRef)
+}
+
+// fromData fetches the public key from raw data and returns the verifier
+func fromData(raw []byte, hashAlgorithm crypto.Hash) (signature.Verifier, error) {
+	pubKey, err := cryptoutils.UnmarshalPEMToPublicKey(raw)
+	if err != nil {
+		return nil, fmt.Errorf("%w: %v", ErrorDecodeKey, err)
+	}
+	v, err := signature.LoadVerifier(pubKey, hashAlgorithm)
+	if err != nil {
+		return nil, fmt.Errorf("%w: %v", errorLoadVerifier, err)
+	}
+	return v, nil
+}
+
+// keyPairSecret fetches the secret from a k8sRef
+func keyPairSecret(ctx context.Context, k8sRef string, k8s kubernetes.Interface) (*v1.Secret, error) {
+	split := strings.Split(strings.TrimPrefix(k8sRef, keyReference), "/")
+	if len(split) != 2 {
+		return nil, errorK8sSpecificationInvalid
+	}
+	namespace, name := split[0], split[1]
+
+	s, err := k8s.CoreV1().Secrets(namespace).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("%w: %v", errorSecretNotFound, err)
+	}
+
+	return s, nil
+}
+
+// hashAlgorithm returns a crypto.Hash code using an algorithm name as input parameter
+func hashAlgorithm(algorithmName string) (crypto.Hash, error) {
```

Comment:
similar feedback to keyPairSecret (and if possible please check other function names for this as well): it is helpful if a function actually takes an action of some kind to include a verb indicating what the action does in the function name, in this case something like 'matchHashAlgorithm' or something